thirdeye-ui/src/app/components/subscription-group-wizard/alerts-dimensions/add-dimension-dialog/add-dimension-dialog.component.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Divider, TextField, Typography } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Alert } from "../../../../rest/dto/alert.interfaces";
+import { useGetEnumerationItems } from "../../../../rest/enumeration-items/enumeration-items.actions";
+import { getMapFromList } from "../../../../utils/subscription-groups/subscription-groups.util";
+import { InputSection } from "../../../form-basics/input-section/input-section.component";
+import { Association } from "../../subscription-group-wizard.interface";
+import { AlertsTable } from "../alert-dimension-table/alert-dimension-table.component";
+import { getAssociationId } from "../alerts-dimensions.utils";
+import { AddDimensionDialogProps } from "./add-dimension-dialog.interface";
+import {
+    getAssociationIdsForAlert,
+    getEnumerationItemsForAlert,
+} from "./add-dimension-dialog.utils";
+
+export const AddDimensionsDialog: FunctionComponent<AddDimensionDialogProps> =
+    ({ alerts, associations, updateAssociations }) => {
+        const [associationsSelected, setAssociationsSelected] = useState<
+            Record<string, boolean>
+        >(
+            Object.assign(
+                {},
+                ...associations.map((item) => ({ [item.id]: true }))
+            )
```

Comment:
Use object spread like the rest of the codebase please

Reply:
`Object.assign` is required to merge all the keys into one object


![image](https://user-images.githubusercontent.com/20799363/220913879-bea7f56e-f9bc-4590-a663-44ce36ae7624.png)